uamqp/aio/_link_async.py

```diff
@@ -122,9 +121,10 @@ async def _set_state(self, new_state):
             _LOGGER.error("Link state change callback failed: '%r'", e, extra=self.network_trace_params)
 
     async def _remove_pending_deliveries(self):  # TODO: move to sender
-        async with create_task_group() as tg:
-            for delivery in self._pending_deliveries.values():
-                await tg.spawn(delivery.on_settled, LinkDeliverySettleReason.NotDelivered, None)
+        futures = []
+        for delivery in self._pending_deliveries.values():
+            futures.append(asyncio.create_task(delivery.on_settled(LinkDeliverySettleReason.NotDelivered, None)))
+        await asyncio.gather(*futures)
```

Comment:
Looks like in general we will need a plan for handling the errors when we gather futures like this.

Reply:
agreed, this is an interesting problem.

I feel maybe we're good not to deal with the error raised here -- that's what we have in our current sync implementation, but I need to keep thinking more on this and how it would affect developers.

Reply:
issue logged: https://github.com/Azure/azure-sdk-for-python/issues/22975